deploy/alerts/terraform/roles/teardown/tasks/main.yml

```diff
@@ -0,0 +1,78 @@
+---
+
+# This play require the following variable set
+#   newrelic_personal_api_key: "<from user config newrelic credential>"
+#   newrelic_admin_api_key: "<from user config newrelic credential>"
+#   newrelic_account_id: "<from user config newrelic credential>"
+#   newrelic_region: "<from user config newrelic credential>"
+#   service_display_name: "The entity display_name to create alert for"
+# The fields below are optionals
+#   s3_bucketname_tfstate: "<an AWS S3 bucket name to use for the terraform state file>"
+#   aws_access_key: the aws api key when an S3 bucket is used
+#   aws_secret_key: the aws secret key when an S3 bucket is used
+#   aws_region: the aws region key when an S3 bucket is used
+
+- debug:
+    msg: "Tearing down alerts for {{ service_display_name }} using Terraform for Golden Signals policy"
+
+- fail:
+    msg: "newrelic_personal_api_key is required. Create this entry in your user config file"
+  when: newrelic_personal_api_key is not defined
+- fail:
+    msg: "newrelic_admin_api_key is required. Create this entry in your user config file"
+  when: newrelic_admin_api_key is not defined
+- fail:
+    msg: "newrelic_account_id is required. Create this entry in your user config file"
+  when: newrelic_account_id is not defined
+- fail:
+    msg: "newrelic_region is required. Create this entry in your user config file"
+  when: newrelic_region is not defined
+- fail:
+    msg: "The S3 bucket name '{{ s3_bucketname_tfstate }}' requires the addition of 'provider_credential': 'aws' in your deploy config file to use the aws credentials to access the bucket."
+  when: s3_bucketname_tfstate is defined and aws_secret_key is not defined
+
+- block:
+  - name: Creating terraform var file
+    set_fact:
+      tfvars_json: "{{ (tfvars_json|default({})) | combine({
+        'name': service_display_name,
+        'service': {
+          'duration': 5,
+          'response_time_threshold': 5,
+          'throughput_threshold': 1,
+          'error_percentage_threshold': 5,
+          'cpu_threshold': 90
+        },
+        'newrelic_api_key': newrelic_personal_api_key,
+        'newrelic_admin_api_key': newrelic_admin_api_key,
+        'newrelic_account_id': newrelic_account_id,
+        'newrelic_region': newrelic_region
+        })
+      }}"
+
+  - name: "write terraform var to file {{playbook_dir}}/tfvars.json"
+    copy:
+      content: "{{ tfvars_json | to_nice_json }}"
+      dest: "{{ playbook_dir }}/tfvars.json"
+
+  - name: create infra config file
+    template:
+      src: ../../templates/main.tf
+      dest: "{{ playbook_dir }}/main.tf"
+
+  - name: "Initialize terraform"
+    shell: "terraform init"
+
+  - name: "Execute terraform destroy"
+    shell: "terraform destroy -auto-approve -var-file={{ playbook_dir }}/tfvars.json"
+    register: terraform_result
+    retries: 8
+    delay: 15
+    until: terraform_result is not failed
```

Comment:
When tearing down terraform fails, it stop the deployer and blocks tearing down anything else.  This step failed because my deploy failed to create the alerts in the first place.
I added 'ignore_errors: true' locally so I could tear down all the cloud resources.

Reply:
I've pushed a fix to handle more gracefully the situation when the s3 bucket no longer exist. This is a bit different than the issue you noted though.
Usually not finding an entity relates to the wrong account is being used between what was provisioned and what is destroyed.